base/src/main/scala/data/ProCompose.scala

```diff
@@ -0,0 +1,54 @@
+package scalaz
+package data 
+
+object ProCompose {
+  import typeclass.{Profunctor, Strong, Category, StrongClass, CategoryClass}
+  import typeclass.Strong._
+  import typeclass.Profunctor._
+
+  trait ProCompose[P[_, _], A, B] { } 
+
+  case class Pure[P[_, _], A, B](pab: A => B) extends ProCompose[P, A, B]
+
+  case class Composed[P[_, _], A, X, B](pab: P[A, X], more: ProCompose[P, X, B]) extends ProCompose[P, A, B]
```

Comment:
> I feel like free categories are a useful addition to Scalaz 8, but if you disagree

No, I agree. I don't see it as a prerequisite for this PR, though.

In fact, `FreeCategory[F :+: Function1]` is slightly redundant for this&mdash;there are two ways to express identity morphisms:
 - `FreeCategory.lift(\/-(identity[A]))`
 - `FreeCategory.identity[A]`

`FreeCompose` would be sufficient.

And `FreeCategory[F]` could be expressed as `FreeCompose[F :+: ===]`. Though using these more generic building blocks costs in extra boxing, so I wouldn't mind occasionally just bake in a specialized version.